src/main/java/org/assertj/core/api/AbstractCharSequenceAssert.java

```diff
@@ -1438,6 +1481,31 @@ public SELF containsPattern(CharSequence regex) {
     return myself;
   }
 
+  /**
+   * Verifies that the actual {@code CharSequence} contains the given regular expression.
+   * Accepts a {@code Consumer<Matcher>} for further verification of the found match.
+   * <p>
+   * Example :
+   * <pre><code class='java'> // assertion will pass
+   * assertThat(&quot;Frodo&quot;).containsPatternSatisfying(&quot;.o(.o)&quot;,
+   *   match -&gt; assertThat(match).isEqualTo(&quot;do&quot;));
+   * </code></pre>
+   *
+   * @param regex the regular expression to find in the actual {@code CharSequence}.
+   * @param matchSatisfies a consumer for further verifying the Matcher.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the given pattern is {@code null}.
+   * @throws PatternSyntaxException if the regular expression's syntax is invalid.
+   * @throws AssertionError if the actual {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the given regular expression cannot be found in the actual {@code CharSequence}.
+   */
+  public SELF containsPatternSatisfying(CharSequence regex, Consumer<Matcher> matchSatisfies) {
+    Matcher matcher = Pattern.compile(regex.toString()).matcher(actual);
+    strings.assertContainsPattern(info, actual, matcher);
+    matchSatisfies.accept(matcher);
+    return myself;
```

Comment:
I think this could simply call `containsPatternSatisfying(Pattern pattern, Consumer<Matcher> matchSatisfies)` as:
```java
return containsPatternSatisfying(Pattern.compile(regex.toString()), matchSatisfies);
```